src/main/java/edu/harvard/iq/dataverse/harvest/client/HarvesterServiceBean.java

```diff
@@ -380,6 +353,68 @@ private Long processRecord(DataverseRequest dataverseRequest, Logger hdLogger, P
 
         return harvestedDataset != null ? harvestedDataset.getId() : null;
     }
+    
+    File retrieveProprietaryDataverseMetadata (String remoteApiUrl) throws IOException {
+        InputStream in;
+        int responseCode = 0;
+        HttpURLConnection con = null;
+        
+        File tempMetadataFile = File.createTempFile("meta", ".tmp");
+        
+        try {
+            URL url = new URL(remoteApiUrl);
+
+            con = (HttpURLConnection) url.openConnection();
+            con.setRequestProperty("User-Agent", "DataverseHarvester/3.0");
+            responseCode = con.getResponseCode();
+        } catch (MalformedURLException mue) {
+            throw new IOException ("Bad API URL: "+remoteApiUrl);
+        } 
+        
+        if (responseCode == 200) {
+            in = con.getInputStream();
+            
+            FileOutputStream tempOut = new FileOutputStream(tempMetadataFile);
+            
+            int bufsize;
+            byte[] buffer = new byte[4 * 8192];
+
+            while ((bufsize = in.read(buffer)) != -1) {
+                tempOut.write(buffer, 0, bufsize);
+                tempOut.flush();
+            }
+
+            in.close();
+            tempOut.close();
+            return tempMetadataFile;
+        }
+
+        throw new IOException("Failed to download extended metadata.");
```

Comment:
IMHO this code is really hacky Java with C origin, also using some Exception anti-patterns.

May I suggest starting to use try-with-resources, considering [`InputStream.transferTo()`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/InputStream.html#transferTo(java.io.OutputStream)) and looking at the modern [Java included HTTP client](https://docs.oracle.com/en/java/javase/11/docs/api/java.net.http/java/net/http/package-summary.html)? (See also https://www.mocklab.io/blog/which-java-http-client-should-i-use-in-2020/)